corehq/apps/userreports/reports/view.py

```diff
@@ -119,3 +150,68 @@ def url_pattern(cls):
         from django.conf.urls import url
         pattern = r'^{slug}/(?P<report_config_id>[\w\-:]+)/$'.format(slug=cls.slug)
         return url(pattern, cls.as_view(), name=cls.slug)
+
+    @property
+    def type(self):
+        return self.prefix
+
+    @property
+    def sub_slug(self):
+        return self.report_config_id
+
+    @classmethod
+    def get_report(cls, domain, slug, report_config_id):
+        report = cls()
+        report.domain = domain
+        report.report_config_id = report_config_id
+        report.name = report.title
+        return report
+
+    @property
+    def report_type(self):
+        return self.type
+
+    @property
+    def url(self):
+        return reverse(self.slug, args=[self.domain, self.report_config_id])
+
+    @property
+    @memoized
+    def export_table(self):
+        try:
+            data = self.data_source
+            data.set_filter_values(self.filter_values)
+        except UserReportsError as e:
+            return self.render_json_response({
+                'error': e.message,
+            })
+
+        report_config = ReportConfiguration.get(self.report_config_id)
+        raw_rows = list(data.get_data())
+        headers = [column['display'] for column in report_config.columns]
+        columns = [column['field'] for column in report_config.columns]
```

Comment:
i'm surprised these hit the report config and not the data source, e.g. https://github.com/dimagi/commcare-hq/blob/scheduled-ucr/corehq/apps/userreports/reports/view.py#L122

is there a reason for that?